src/audio/dai-zephyr.c

```diff
@@ -977,8 +977,9 @@ int dai_common_params(struct dai_data *dd, struct comp_dev *dev,
 			return err;
 		}
 	} else {
+		/* allocate not shared buffer */
 		dd->dma_buffer = buffer_alloc(buffer_size, SOF_MEM_CAPS_DMA, 0,
-					      addr_align);
+					      addr_align, false);
```

Comment:
We actually never used coherent.h to mark the handovers between DSP cores and DMA engines (like is done e.g. with DMA API in Linux -> https://www.kernel.org/doc/Documentation/DMA-API-HOWTO.txt ).

So while it's a bit confusing, I think this is good. I wonder we should reflect this in the code comments and other documentation that "not shared buffer" is actually "not shared buffer descriptor". The logic driven by whether object is shared or not, deals with buffer metadata and not the actual payload buffers. For casual observer, "shared buffer" certainly implies the statements here apply to the buffer itself (while that really is not the case here).

Reply:
> The logic driven by whether object is shared or not, deals with buffer metadata and not the actual payload buffers. For casual observer, "shared buffer" certainly implies the statements here apply to the buffer itself (while that really is not the case here).

@kv2019i really? I actually think it's both. A buffer is marked as "shared" if it's between two components, executed on different cores. So, if the producer runs on core A, that core A will both access the buffer control structure and actually write data into the buffer. Then consumer on core B will access the control structure and read data on core B.